### PR TITLE
Force RGBA16 when scene filtering is active

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
C# side fix for https://fogbugz.unity3d.com/f/cases/1228736/ 

depends on https://ono.unity3d.com/unity/unity/pull-request/119986/_/graphics/scene-filtering-to-public (21.2)

For upcoming changes, we need to make sure that when scene filtering is active we output a correct alpha. This PR allows that by forcing the format to be FP16 when scene filtering is active. 

This also required some small change to the realloc of the color pyramid and when the post process checks for alpha support. 

Note: the C++ needed change landed in 08e7e027f658 

Unfortunately cannot be really tested without merging the C++ branch mentioned in the fogbugz and trunk. But checked locally with a tessellated sphere and all works correctly (while it wouldn't before) 

image

### CODE REVIEW

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
todo: need to retarget to master when hd/bugfix is merge. This PR depends on C++ only in 2021.2

**Review by sebastienlagarde — APPROVED** [review]

File: com.unity.render-pipelines.high-definition/Runtime/Compositor/CompositionManager.cs [hunk 1]
```diff
@@ -192,11 +192,11 @@ bool ValidatePipeline()
             if (hdPipeline != null)
             {
                 m_AlphaSupport = AlphaChannelSupport.RenderingAndPostProcessing;
-                if (hdPipeline.asset.currentPlatformRenderPipelineSettings.colorBufferFormat == RenderPipelineSettings.ColorBufferFormat.R11G11B10)
+                if (hdPipeline.GetColorBufferFormat() == (GraphicsFormat)RenderPipelineSettings.ColorBufferFormat.R11G11B10)
                 {
                     m_AlphaSupport = AlphaChannelSupport.None;
                 }
-                else if (hdPipeline.asset.currentPlatformRenderPipelineSettings.postProcessSettings.bufferFormat == PostProcessBufferFormat.R11G11B10)
+                else if (hdPipeline.GetColorBufferFormat() == (GraphicsFormat)PostProcessBufferFormat.R11G11B10)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Compositor/CompositionManager.cs:199]: Is it intended that we don't check the postprocess color format anymore here?

Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Compositor/CompositionManager.cs:199]: Oh no, I think is a copy paste error.

**Comment by sebastienlagarde** [comment]
This PR depends on another C++ PR that the one in description. It will be done mid next week.

**Comment by FrancescoC-unity** [comment]
https://github.com/Unity-Technologies/Graphics/pull/3975 ported here

**Pull request closed without merging**